pkg/maps/ipcache/ipcache.go

```diff
@@ -248,3 +250,21 @@ var (
 func Reopen() error {
 	return IPCache.Map.Reopen()
 }
+
+// Function to update IPCache map with VTEP CIDR
+func UpdateIPCacheVTEPMapping(newCIDR *cidr.CIDR, newTunnelEndpoint net.IP,
+	securityIdentity uint32, encryptKey uint8) error {
+
+	key := NewKey(newCIDR.IP, newCIDR.Mask)
+
+	value := RemoteEndpointInfo{
+		SecurityIdentity: securityIdentity,
+		Key:              encryptKey,
+	}
+	if ip4 := newTunnelEndpoint.To4(); ip4 != nil {
+		copy(value.TunnelEndpoint[:], ip4)
```

Comment:
Should this fail or log an error if it's not an IPv4 address? Perhaps could check that in `validateVTEP` below? Since you have follow-up PRs coming, lets not fix it in this one.

Reply:
Actually yeah I missed this the first time. So the problem @joamaki is pointing out is that if `ip4 == nil` (eg because the user specified an IPv6 address), then we just won't initialize the `value` properly for the peer IP but we will still update the ipcache. This is a bit weird, since the VTEP feature won't be achieving anything by inserting such a half-initialized value. 

And indeed from a brief look, the `validateVTEP()` function doesn't seem to ensure that the IP addresses are IPv4.

Could you either ensure that these IPs are IPv4 in `validateVTEP()` or return an error from this case where `ipv4 == nil`?

Reply:
ok, I will validate IPv4 address in `validateVTEP()`